### PR TITLE
Minor cleanup for Astrometric

### PR DESCRIPTION
**Pull request by cdeil**
This PR does some minor cleanup (typos, whitespace) for the Astrometric frame code recently added by @alexrudy and @eteq .

@alexrudy @eteq - How about adding the `rotation` option to [SkyCoord.astrometric_frame](http://docs.astropy.org/en/latest/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord.astrometric_frame) and pass it along to [AstrometricFrame](http://docs.astropy.org/en/latest/api/astropy.coordinates.AstrometricFrame.html)? Would it be OK with you if I add that as a convenience?
I think it would be nice, but if it's controversial I'll just leave this PR as-is.

### CODE REVIEW

**Comment by cdeil** [comment]
Looks like the `from __future__ import unicode_literals` lead to this error on Python 2:
https://travis-ci.org/astropy/astropy/jobs/134089896#L2088

Putting this in every file is part of the Astropy coding guidelines (see [here](http://docs.astropy.org/en/latest/development/codeguide.html)).
Should it be omitted for this file? Or does someone know how to modify the `AstrometricFrame` class implementation to make it work with `from __future__ import unicode_literals` on Python 2?

**Comment by cdeil** [comment]
This is the error that happens on Python 2 for this branch, where `from __future__ import unicode_literals` was added to `astrometric.py`:
```python
>>> from astropy.coordinates import AstrometricFrame, SkyCoord
>>> origin = SkyCoord(1, 2, unit='deg')
>>> AstrometricFrame(origin=origin)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "astropy/coordinates/builtin_frames/astrometric.py", line 197, in __new__
    newcls = make_astrometric_cls(origin_frame.__class__)
  File "astropy/coordinates/builtin_frames/astrometric.py", line 108, in make_astrometric_cls
    {'__doc__': AstrometricFrame.__doc__})
  File "astropy/coordinates/builtin_frames/astrometric.py", line 71, in __new__
    res = super(AstrometricMeta, cls).__new__(cls, newname, bases, members)
  File "astropy/coordinates/baseframe.py", line 77, in __new__
    tmp_cls = super(FrameMeta, mcls).__new__(mcls, name, bases, members)
TypeError: type() argument 1 must be string, not unicode
```
I don't know how to resolve it.
Does anyone have a suggestion?
(For now, I'll remove `from __future__ import unicode_literals` again so that this PR can move forward.)

**Comment by cdeil** [comment]
There were some comments in the code left that talked about adding "d" to the "lon" and "lat" names, which is no longer the case. I tried to fix the comments in 9505058ee72e10dc5da24293cc9700ec65d5707e, but @eteq and @alexrudy, could you please check the comments and code in `AstrometricMeta.__new__` that handles the component names if it's now correct?

**Comment by cdeil** [comment]
I've gone ahead and added the rotation option to `SkyCoord.astrometric_frame` in 0a39669.

**Comment by taldcroft** [comment]
> Should it be omitted for this file? Or does someone know how to modify the AstrometricFrame class implementation to make it work with from __future__ import unicode_literals on Python 2?

It can be omitted.  I think the use of `unicode_literals` is more controversial than the guideline would suggest.  `io.fits` and `io.ascii` do not use them.  I personally would advocate not using that in code but testing with unicode literals (and str literals) in Py2 as needed.

File: astropy/coordinates/sky_coordinate.py [hunk 1]
```diff
@@ -1027,18 +1027,23 @@ def position_angle(self, other):
...
         Returns the astrometric frame with this `SkyCoord` at the origin.
 
         Returns
         -------
-        astrframe: AstrometricFrame
+        astrframe : `~astropy.coordinates.AstrometricFrame`
             An astrometric frame of the same type as this `SkyCoord` (e.g., if
             this object has an ICRS coordinate, the resulting frame is
             AstrometricFrame with an ICRS origin)
+        rotation : Quantity with angle units
```
Comment (mhvk) [inline astropy/coordinates/sky_coordinate.py]: Maybe just `~astropy.coordinates.Angle`? Could still add "or `~astropy.units.Quantity` with angle units"

**Comment by mhvk** [comment]
Agree to just omit the `unicode_literals`. One tiny comment as you would be editing anyway. Otherwise, this looks all ready to go in.

File: astropy/coordinates/sky_coordinate.py (hunk 1 above)
Comment (cdeil) [inline astropy/coordinates/sky_coordinate.py]: I copy & pasted this from the `AstrometricFrame` docstring. Do you want me to change it in both places, just here, or leave as is?

Comment (mhvk) [inline astropy/coordinates/sky_coordinate.py]: Definitely both should be the same. And I think it is good practice to link to classes (like `~astropy.units.Quantity`). But this is not very important, so I'll leave the decision up to you.

**Comment by cdeil** [comment]
@mhvk - Thanks for the feedback. I applied your inline comment and also squashed the commits.

**Comment by alexrudy** [comment]
@cdeil I tried with 'unicode_literals', and I think that we just need to convert the frame name to a native string type, since it becomes a class name. 

See https://github.com/alexrudy/astropy/commit/09cc02f894be630228bcb9bb9efd412147726619

This issue is one of the reasons people aren't in favor of 'unicode_literals' any more, as it might break if someone ever got a non-ascii character in there. But I don't think we are going to be creating frames with non-ascii names any time soon. So I'm 0 on whether to leave 'unicode_literals' in or take it out. Otherwise LGTM

**Comment by cdeil** [comment]
>  So I'm 0 on whether to leave 'unicode_literals' in or take it out. 

Let's leave as-is then.

I think this PR is non-controversial and ready to go then.

**Comment by mhvk** [comment]
It all looks good, so I merge this before the name change in promised in #5002 makes this unmergeable!

**Pull request merged**